src/pages/Lending/Pool/components/Repay/RepayConfirm.tsx

```diff
@@ -195,6 +208,36 @@ export const RepayConfirm = ({
       contractAddress: undefined,
     })
 
+    if (repaymentAsset.assetId === thorchainAssetId) {
+      const { account } = fromAccountId(repaymentAccountId)
+
+      const adapter = chainAdapter as unknown as thorchain.ChainAdapter
+
+      // repayment using THOR is a MsgDeposit tx
+      const { txToSign } = await adapter.buildDepositTransaction({
```

Comment:
Figured out this is a better one to use instead of manually building the `StdSignTx` as we do with swapper - adapter already does the heavy duty lifting for us